torch/fx/proxy.py

```diff
@@ -104,6 +104,61 @@ def __init__(self, graph: Graph):
 class TraceError(ValueError):
     pass
 
+
+def _create_friendly_names(args: Tuple[Any, ...], kwargs: Dict[str, Any], frames_up : int):
+    """
+    Given an args/kwargs object, go through and try to pull out the names for
+    each contained Proxy from the Python interpreter frame `frames_up` above
+    us in the stack. If found, assign that name to the Proxy's underlying Node's
+    unique name
+    """
+    frame = inspect.currentframe()
+    if not frame:
+        raise RuntimeError("failed to inspect frame")
+
+    i = 0
+    while i < frames_up + 1:
+        frame = frame.f_back
+        if not frame:
+            raise RuntimeError("failed to get frame")
+        i += 1
+
+    f_locals = frame.f_locals
+
+    def _assign_friendly_name(a : Any):
+        # aggregates
+        if isinstance(a, (tuple, list)):
```

Comment:
I don't like the fact that this has to iterate the structure of our arguments. Now it exists in _3_ places: map_arg, create_arg, and here. This logic should probably go into create_arg where the conversion of a proxy to a node is handled. It should also probably be a function on the tracer e.g. `self.find_reasonable_name(proxy)`